### PR TITLE
RDK-35544: HdmiInput thunder plugin changes for ALLM method/event

### PR DESCRIPTION
**Pull request by dhivya-ilangovan**
Reason for change: added code for getSupportedGameFeatures,
getHdmiALLMStatus
and the event HdmiALLMStatusUpdate
Test Procedure: available in Jira
Risks: Low
Signed-off-by: Dhivya Ilangovan <dhivya.dilangovan@sky.uk>

### CODE REVIEW

File: HdmiInput/HdmiInput.cpp [hunk 1]
```diff
@@ -42,11 +42,14 @@
 #define HDMIINPUT_METHOD_START_HDMI_INPUT "startHdmiInput"
 #define HDMIINPUT_METHOD_STOP_HDMI_INPUT "stopHdmiInput"
 #define HDMIINPUT_METHOD_SCALE_HDMI_INPUT "setVideoRectangle"
+#define HDMIINPUT_METHOD_SUPPORTED_GAME_FEATURES "getSupportedGameFeatures"
+#define HDMIINPUT_METHOD_ALLM_STATUS "getHdmiALLMStatus"
```
Comment (apatel859) [inline HdmiInput/HdmiInput.cpp:46]: If ALLM is one of the same Feature and if you have multiple game feature should not this APIO should be generic to take Game mode as input param and return its status?

File: HdmiInput/HdmiInput.cpp [hunk 2]
```diff
@@ -42,11 +42,14 @@
...
 
 #define HDMIINPUT_EVENT_ON_DEVICES_CHANGED "onDevicesChanged"
 #define HDMIINPUT_EVENT_ON_SIGNAL_CHANGED "onSignalChanged"
 #define HDMIINPUT_EVENT_ON_STATUS_CHANGED "onInputStatusChanged"
 #define HDMIINPUT_EVENT_ON_VIDEO_MODE_UPDATED "videoStreamInfoUpdate"
+#define HDMIINPUT_EVENT_ON_ALLM_CHANGED "hdmiALLMStatusUpdate"
```
Comment (apatel859) [inline HdmiInput/HdmiInput.cpp:52]: same as above comment

**Comment by Firdose2109** [comment]
Can one of the admins verify this patch?

**Comment by anand-ky** [comment]
Closing older PRs. Pls resubmit to latest 2205 sprint.

**Pull request closed without merging**